File: quarkus/services/consumer/src/main/java/io/mendirl/quarkus/services/consumer/SchedulingConfiguration.java
package io.mendirl.quarkus.services.consumer;

import io.quarkus.scheduler.Scheduled;
import org.eclipse.microprofile.rest.client.inject.RestClient;
import org.jboss.logging.Logger;

import javax.enterprise.context.ApplicationScoped;
import javax.inject.Inject;
import java.util.concurrent.TimeUnit;

@ApplicationScoped
public class SchedulingConfiguration {

    @Inject
    Logger logger;

    @Inject
    @RestClient
    private ProducerClient client;

    @Scheduled(delay = 1, delayUnit = TimeUnit.SECONDS, every = "1s")
    public void scheduledWeb() {
        var position = client.position();
//            .subscribe {
        logger.infof("position retrieved: %s", position);
//        }
    }
}

File: spring/services/consumer/src/main/kotlin/io/mendirl/spring/services/consumer/SchedulingConfiguration.kt
package io.mendirl.spring.services.consumer

import org.slf4j.LoggerFactory
import org.springframework.context.annotation.Configuration
import org.springframework.scheduling.annotation.EnableScheduling
import org.springframework.scheduling.annotation.Scheduled

@EnableScheduling
@Configuration
class SchedulingConfiguration(
    val client: ProducerClient
) {

    private val logger = LoggerFactory.getLogger(SchedulingConfiguration::class.java)

    @Scheduled(initialDelay = 1000, fixedRate = 1000)
    fun scheduledWeb() {
        client.position()
            .subscribe {
                logger.info("position retrieved: $it")
            }
    }

    @Scheduled(initialDelay = 1000, fixedRate = 1000)
    fun scheduledRSocket() {
        client.message()
            .subscribe {
                logger.info("message retrieved: $it")
            }
    }
}
